hangman/Hangman.py

```diff
@@ -0,0 +1,63 @@
+from operator import truediv
+import random
+from typing import List
+from word_generator import WordGenerator
+from constants import *
+
+
+class Hangman:
+    def __init__(self, word: str, hints_left, chances_left) -> None:
+        self.curr_word: str = word.upper()
+        self.dashed_word: List = [UNGUESSED_CHAR] * len(word)
+        self.chances_left: int = chances_left
+        self.hints_left: int = hints_left
+        self.guessed_alphabets: List[bool] = [False] * 26
+        self.game_over = False
+
+    def check_game_over(self) -> bool:
+        if(UNGUESSED_CHAR not in self.dashed_word):
+            print("YOU HAVE WON!!! The word is", self.curr_word)
+            return True
+        elif(self.chances_left >= 0):
+            return False
+        else:
+            print("GAME OVER!!! The word is", self.curr_word)
+            return True
+
+    def hint(self) -> bool:
+        if self.hints_left == 0:
+            print(self.hints_left, "Hints left!!!")
+            return False
+
+        while(True):
```

Comment:
the logic is flawed no ? we need to show all places for that characters that we show hint for

Reply:
Manish mentioned that we have to show only one place for that character so that user should still guess if there are any other occurrences of that character.